### DISCUSSION TITLE
Should run-many run targets in specified order?

### DISCUSSION THREAD

**Discussion by dmcweeney** [nrwl/nx#26008] [Questions] [answered]
I noticed that when you run the run-many --targets=lint,test,build --projects=*-app
that the targets are executed in random order on a random project order.

I would have considered that the targets should be executed in the specified order?

Would this be considered a bug?

**Comment by jaysoo** [comment, marked as answer]
Hi @dmcweeney , this isn't a bug, as Nx schedules tasks depending on various factors, including dependent tasks, distribution, etc.

If you need to run tasks sequentially, I recommend breaking them out into separate commands.  e.g. `nx run-many -t lint && nx run-many -t test && nx run-many -t build`

**Comment by dmcweeney** [comment]
Thanks @jaysoo thats what I suspected! I've broken out the tasks as suggested.
